pkg/gadgets/seccomp/gadget.go

```diff
@@ -183,6 +184,96 @@ func seccompProfileAddLabelsAndAnnotations(
 	}
 }
 
+type SeccompProfileNsName struct {
+	namespace string
+	name      string
+
+	// generateName indicates whether the name field has to be used as
+	// resource's Name or GeneratedName
+	generateName bool
+}
+
+// getSeccompProfileNextName computes the next profile name that has to be used
+// for a specific podname given a SeccompProfile list. This function returns:
+// podName: If there do not exist profiles with podname or podname-X as name.
+// podName-2: If there exist a profile with the podname but no one with podname-X.
+// podName-<X+1>: If there exist at least one profile with podname-X.
+func getSeccompProfileNextName(profileList []seccompprofilev1alpha1.SeccompProfile, podName string) string {
```

Comment:
When we discussed this, I did not see the implementation like this.
Maybe we can have a global map which key is pod name and value is the last generated profile identifier (_i.e._ the `X`).
Thus, we access the map, get the last value, format the name and increment the value.

Nonetheless, your solution has the benefit to reuse identifier.
If we had `pod-1` and `pod-42` but for profile `pod-42` was deleted, then next profile name would be `pod-2`.

What do you think?

Reply:
IMO, if I am running different tests and I remove the seccomp profiles I generated before, it's because I want to reuse that name. Let's see other thoughts: @alban @mauriciovasquezbernal

Reply:
> Maybe we can have a global map which key is pod name and value is the last generated profile identifier (i.e. the X).

Could you extend a bit more in this idea please? Where will be this map stored? 

> if I am running different tests and I remove a seccomp profile I generated before, it's because I want to reuse that name 

I agree. I don't see how we could do it differently.

Reply:
My idea was to write something like this:

```golang
// Pseudocode
var profileNames = map[string][int]

func (profileName string) {
  value, ok := profileNames[profileName]
  if !ok {
    profilesNames[profileName] = 0
    return profileName
  }
  profilesNames[profileName]++
  return fmt.Sprintf("%s-%d", profileName, value)
}
```

As this code will live under `/pkg/gadgets` it would be possible to have a global map as it will live as long as the gadget container.
The problem with this solution is we need to synchronize this maps for each gadget pods we have (basically one per node) and this would be complicated.
So, your solution is the best of both world.